src/devtools/client/debugger/src/actions/event-listeners.js

```diff
@@ -17,6 +19,36 @@ export async function setupEventListeners(_, store) {
   await store.dispatch(setEventListeners(eventListeners));
 }
 
+async function fetchEventTypePoints(categories) {
+  const eventTypes = categories
+    .map(cat => cat.events)
+    .flat()
+    .map(e => e.id);
+
+  const sessionId = await ThreadFront.waitForSession();
+
+  const eventTypePoints = await Promise.all(
+    eventTypes.map(
+      eventType =>
+        new Promise(async resolve => {
+          analysisManager.runAnalysis(
+            {
+              sessionId,
+              mapper: `return [{ key: input.point, value: input }];`,
+              effectful: false,
+              eventHandlerEntryPoints: [{ eventType }],
+            },
+            {
+              onAnalysisPoints: points => resolve([eventType, points]),
+            }
+          );
+        })
+    )
+  );
+
+  return Object.fromEntries(eventTypePoints);
```

Comment:
I'm pretty happy with how compact this is. I'd like to add this behind a feature flag while we're experimenting